File: app/src/main/java/com/szollosi/sensormusicplayer/MyDataPoint.java
package com.szollosi.sensormusicplayer;

import androidx.annotation.NonNull;

import com.github.mikephil.charting.data.Entry;

public class MyDataPoint extends Entry implements Comparable<MyDataPoint> {

  MyDataPoint(float x, float y) {
    super(x, y);
  }

  @Override
  public int compareTo(@NonNull MyDataPoint o) {
    return (int) (Math.abs(this.getY()) - Math.abs(o.getY()));
  }
}

File: app/src/main/java/com/szollosi/sensormusicplayer/MyDataPoint.kt
package com.szollosi.sensormusicplayer

import com.github.mikephil.charting.data.Entry
import kotlin.math.abs

class MyDataPoint internal constructor(x: Float, y: Float) : Entry(x, y), Comparable<MyDataPoint> {

    override fun compareTo(other: MyDataPoint): Int {
        return (abs(this.y) - abs(other.y)).toInt()
  }
}
